tensorflow/compiler/tf2tensorrt/convert/ops/fill_ops.cc

```diff
@@ -132,58 +132,72 @@ class ConvertRange : public ConvertFillBase<ConvertRange> {
     const auto& inputs = params.inputs;
     const auto& node_def = params.node_def;
 
-    if (!all_same_types(inputs)) {
-      return errors::InvalidArgument(convert_range_expected_msg(node_def),
-                                     "passed as weights OR tensors");
+    float param[3];
+    all_weights_ = all_integers_ = true;
+    for (int i = 0; i < 3; i++) {
+      const auto& input = inputs.at(i);
+      all_integers_ &= input.TrtDType() == nvinfer1::DataType::kINT32;
+      if (input.is_weights()) {
+        switch (input.TrtDType()) {
+          case nvinfer1::DataType::kFLOAT:
+            param[i] = get_input_param<float>(input);
+            break;
+          case nvinfer1::DataType::kHALF:
+            param[i] = get_input_param<Eigen::half>(input);
+            break;
+          default:  // nvinfer1::DataType::kINT32:
+            param[i] = get_input_param<int>(input);
```

Comment:
```suggestion
          case nvinfer1::DataType::kINT32:
            param[i] = get_input_param<int>(input);
          default:
            return errors::InvalidArgument("Unsopported data type");
```

Reply:
Done. I also added a corresponding test for unsupported data types.